### PR TITLE
Use GitHub API to fetch all player actions in 1 API call

### PR DESCRIPTION
**Pull request by SimonMiaou**
Using the API should also allow us to always have the last version of the `act` file instead of having issues with caching.

Also move the logic of kicking to deal with edge case of repository being removed

To be able to use this change, you need to set the environment variable `GITHUB_TOKEN`, you can generate a token there: https://github.com/settings/tokens

@programical You might want to adjust the sleep as the script ran ~20s faster on my machine after this change

### CODE REVIEW

**Comment by programical** [comment]
Can you prove that the caching issues are still present? This was supposed to be fixed in #47.

**Comment by SimonMiaou** [comment]
I will try as it was more an observation of my BOT not going the direction of my `act` file.

But except caching, this also improve the server speed, and make sure you capture the `act` file of all players at the same time and not in a window of 20s (actually, that may be the cause of the delays of actions instead of caching)

And this also answer the question you had in the original issue:
> Perhaps we could make API requests for the content? I'm not sure if that's supported.

**Comment by SimonMiaou** [comment]
Actually, I start thinking the issue might not be caching but the time the server take to get the inputs.

Looking into providing more informations

**Comment by SimonMiaou** [comment]
Here is some timeline recording from my bot showing that we either have a cache issue or the server is too slow to receive last desired actions:
- [At 13:36:48](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133648-pulling-latest-gitland-changes) Pulling latest changes from gitland
- [At 13:36:49](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133649-pulled-latest-gitland-changes) BOT decide to move right
- [At 13:37:21](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133721-pulling-latest-gitland-changes) Pulling latest changes from gitland, I went right
- [At 13:37:23](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133723-pulled-latest-gitland-changes) BOT decide to move up
- [At 13:37:23](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133723-pulled-latest-gitland-changes) BOT decide to move up
- [At 13:37:57](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133757-pulling-latest-gitland-changes) Pulling latest changes from gitland but didn't change yet
- [At 13:38:30](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133830-pulling-latest-gitland-changes) Pulling latest changes from gitland, I went right again
- [At 13:38:32](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133832-pulled-latest-gitland-changes) BOT decide to move down
- [At 13:39:06](https://github.com/SimonMiaou/gitland-client/blob/master/report-using-old-move.md#133906-pulling-latest-gitland-changes) Pulling latest changes from gitland, I went up

**Comment by programical** [comment]
This is likely related to (and may be solved by) #78.

**Comment by SimonMiaou** [comment]
Then wouldn't it be better to be 1 API call to fetch all actions than 1 API per player? Even concurrently.

File: server.py [hunk 1]
```diff
@@ -151,17 +151,17 @@ def movePlayer(self, playerToMove: str, x: int, y: int):
...
+        query = "{\n"
+        i = 0
+        for player in os.listdir("players"):
+            if os.path.isdir("players/" + player):
+                query = query + "player" + str(i) + ": repository(owner: \"" + player + "\", name: \"gitland-client\") {\nowner {\nlogin\n}\ncontent: object(expression: \"master:act\") {\n... on Blob {\ntext\n}\n}\n}\n"
+                i = i + 1
+        query = query + "}"
+        token = os.environ.get('GITHUB_TOKEN') # https://github.com/settings/tokens
+        request = requests.post('https://api.github.com/graphql', json={'query': query}, headers={"Authorization": "Bearer "+token})
+        return request.json()['data'].values()
```
Comment (programical) [inline server.py:164]: Needs exception handling for a case where the API response doesn't contain a `data` key.

File: server.py [hunk 2]
```diff
@@ -176,50 +176,47 @@ def updateGameState(self):
                 x += 1
             y += 1
 
-        # add players
+        # player actions
+        for entry in self.fetchPlayerActions():
+            player = entry['owner']['login']
+            action = entry['content']['text'].strip()
```
Comment (programical) [inline server.py:182]: Needs exception handling for unexpected content and missing values.

File: server.py [hunk 3]
```diff
@@ -176,50 +176,47 @@ def updateGameState(self):
...
-                elif action == "right":
-                    self.movePlayer(player, x + 1, y)
-                elif action == "up":
-                    self.movePlayer(player, x, y - 1)
-                elif action == "down":
-                    self.movePlayer(player, x, y + 1)
-                elif action == "idle":
-                    self.log(player + " didn't do anything")
-                else:
-                    if time.time() - lastActive > 86400: # 24h
```
Comment (programical) [inline server.py:203]: Players with no valid command should still be kicked after 24 hours.

**Comment by SimonMiaou** [comment]
As master changed, it will be easier for me to continue in #102 instead of manage conflict in this PR

**Pull request closed without merging**